src/createAnimatedComponent.js

```diff
@@ -104,7 +104,10 @@ export default function createAnimatedComponent(Component) {
       const viewTag = findNodeHandle(node);
 
       for (const key in this.props) {
-        const prop = this.props[key];
+        let prop = this.props[key];
+        if (prop?.current && prop.current instanceof WorkletEventHandler) {
```

Comment:
We already have an instanceof check for this type few lines below. What's the point of adding another one if we could just reuse the one that's thre and avoid making two identical checks one after another. Also there is no need to check whether prop.current is null before instanceof check as null will never match the type provided.

```
if (prop instanceof AnimatedEvent) {
  prop.attachEvent(node, key);
} else if (prop?.current instanceof WorkletEventHandler) {
  prop.current.registerForEvents(viewTag, key);
}
```

Reply:
this is fixed in #1556